### PR TITLE
Add "Other..." option for checkbox & list questions

### PR DESCRIPTION
**Pull request by WilliamVenner**
This PR adds a feature to checkbox and list questions which allows users to enter custom input. Just need to set `other=True` when creating the question.

https://user-images.githubusercontent.com/14863743/193604405-effc7732-6c69-4e16-9fe0-2ebf7d582f79.mov

### CODE REVIEW

**Comment by Cube707** [comment]
I like the idea and it makes sense as an opt-in feature so that developters an enable it where needed. I will review the code later...

I would argue that its better placed as the last element in the list rather than the first thought. Makes more sense to me.

**Comment by cpiemontese** [comment]
A little bump to know if this has been reviewed yet :pray:

File: src/inquirer/questions.py [hunk 1]
```diff
@@ -32,8 +33,23 @@ def __ne__(self, other):
...
+        choices=None,
+        default=None,
+        ignore=False,
+        validate=True,
+        show_default=False,
+        other=False,
+    ):
+        if other:
+            choices = choices or []
+            choices.append(GLOBAL_OTHER_CHOICE)
```
Comment (Cube707) [inline src/inquirer/questions.py:49]: this is a little redudent, move the `if` doen and append directly to `self._choises`, no need for the additional `or []`

File: src/inquirer/questions.py [hunk 2]
```diff
@@ -32,8 +33,23 @@ def __ne__(self, other):
...
+
         self.name = name
+        self._other = other
```
Comment (Cube707) [inline src/inquirer/questions.py:52]: I would find it more logical, if they would match the order in which they are passed to the function, so maybe move it down?

**Review by Cube707 — CHANGES_REQUESTED** [review]
made some small comments.

And the tests fail, I don'T have the time to check why at the moment though, maybe you can figure it out quicker than i can

File: tests/acceptance/test_checkbox.py [hunk 3]
```diff
@@ -82,6 +82,41 @@ def test_out_of_bounds_down(self):
...
+
+    def test_other_input(self):
+        self.sut.send(key.UP)
+        self.sut.expect(r"\+ Other\.\.\..*", timeout=1)
+        self.sut.send(key.SPACE)
+        self.sut.expect(r": ", timeout=1)
+        self.sut.send("Hello world")
+        self.sut.expect(r"Hello world.*", timeout=1)
+        self.sut.send(key.ENTER)
+        self.sut.expect(r"> X Hello world[\s\S]*\+ Other\.\.\..*", timeout=1)
```
Comment (staticdev) [inline tests/acceptance/test_checkbox.py:99]: the tests are still checking it there are ... aftrr Other. This is why it is failing.

**Review by staticdev — CHANGES_REQUESTED** [review]
Last commit broke the tests. I pointed in comment why.

**Comment by WilliamVenner** [comment]
Cheers all, have made the requested changes. Sorry if the tests still don't pass, I would run them locally but it's a bit of a PITA due to M1 chip. Fingers crossed 🤞🏻

**Review by Cube707 — APPROVED** [review]

**Pull request merged**